tests/components/sensor/test_recorder.py

```diff
@@ -136,8 +136,9 @@ def set_time_zone():
         ("weight", "oz", "oz", "oz", "mass", 13.050847, -10, 30),
     ],
 )
-def test_compile_hourly_statistics(
-    hass_recorder: Callable[..., HomeAssistant],
+async def test_compile_hourly_statistics(
+    recorder_mock: Recorder,
+    hass: HomeAssistant,
```

Comment:
Please start with `hass`.

Reply:
That's not possible, the recorder_mock must be set up before hass.

Reply:
If I recall correctly there are some recorder tests that need to be setup before the `hass` fixture, and this might be one of those cases. Erik will know for sure though.

Reply:
The `recorder_mock` is using the `hass` fixture, so it can't be one of those.

Reply:
The reason why we can't change it is explained here: https://github.com/home-assistant/core/pull/116448#discussion_r1588374159